security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpBasicAuthProvider.java

```diff
@@ -263,11 +267,16 @@ private AuthenticationResponse invalidUser() {
         // extracted to method to make sure we return the same message for invalid user and password
         // DO NOT change this - it is a security problem if the message differs, as it gives too much information
         // to potential attacker
-        return fail("Invalid username or password");
+        return failOrAbstain("Invalid username or password");
     }
 
-    private AuthenticationResponse fail(String message) {
-        return AuthenticationResponse.builder()
+    private AuthenticationResponse failOrAbstain(String message) {
+        return optional ? AuthenticationResponse.builder()
```

Comment:
Please do not use ternary expression for multiline choices. 
This is OK: `optional ? abstain(message) : fail(message)`
Or just use `if (optiona) {...`

Reply:
Replaced ternary  with if/else